### DISCUSSION TITLE
Px proxy "Connection Refused" (Windows 10 with WSL1)

### DISCUSSION THREAD

**Discussion by moyscode** [genotrance/px#163] [General]
I run a Px instance on my Windows 10 with the following content in my `px.ini` file

```
[proxy]
server = company-proxy-sl.company.com:8080
listen = 
port = 3128
gateway = 0
hostonly = 1
allow = 
noproxy = 127.0.0.*,10.*.*.*,192.168.*.*,localhost*
useragent = 
pac = 
username = 
auth = 

[settings]
workers = 3
threads = 8
idle = 30
socktimeout = 20.0
proxyreload = 60
foreground = 0
log = 1
```

Px is working fine. Now I try to connect to the server has from within WSL (Ubuntu 16.04).
Within WSL, I have `~/.bashrc` file with the content as below
`
export http_proxy="http://localhost:3128"
export HTTP_PROXY="http://localhost:3128"
export https_proxy="http://localhost:3128"
export HTTPS_PROXY="http://localhost:3128"
export no_proxy="localhost,127.0.0.1,::1,localhost*"
export NO_PROXY="localhost,127.0.0.1,::1,localhost*"
`

All my proxy env vars (http_proxy, https_proxy, HTTP_PROXY, HTTPS_PROXY) are set to `http://localhost:3128`.

However, when I run `curl www.google.com` on WSL, I get the error `curl: (7) Failed to connect to localhost port 3128: Connection refused`

My colleagues have the same configuration & everything works for them!

**Comment by genotrance** [comment]
Does Px show any connection from curl in the debug output? Is it Px that is refusing the connection or does it not even get the request?

I've not used WSL but wondering if there's any firewall involvement between the Windows and Linux sides.

**Comment by genotrance** [comment]
`—hostonly` already overrides listen so you might want to disable that. Also, on the command line, you just set `—listen=127.0.5.1`.

**Reply by moyscode** [reply]
Oh. OK. I made the change. But in debug mode. I get `Client not allowed: 127.0.0.1` (Pic below) after I start Px in debug mode
![image](https://user-images.githubusercontent.com/92194668/181418371-49e7a0d6-cae7-4de0-b16a-ba41fa5d45ea.png)

**Reply by genotrance** [reply]
So are you in host only mode or did you set listen?

**Reply by genotrance** [reply]
Also, did the request come from windows or WSL?

**Reply by moyscode** [reply]
I am in `hostonly` mode. I did not make any requests at all. From either Windows or WSL. Those `Client not allowed: 127.0.0.1` very generating by themselves.

**Reply by genotrance** [reply]
Well something is trying to access Px - its likely coming cause you have set http_proxy in your bashrc and something is picking it up.

I'll suggest removing the http_proxy entry and testing with `curl --proxy localhost:3128 www.google.com` for now from both Windows and WSL and see whether Px gets the requests first. Try this with `--hostonly`.

Next, disable host only and see if the Px default works (listening to localhost only). Next, you can try adding a specific `--listen`, then followed by an explicit `--allow` list as well. Basically, build your own host only mode. There might be a bug somewhere.
